vital/video_metadata/tests/test_metadata_pos_io.cxx

```diff
@@ -102,38 +106,36 @@ IMPLEMENT_TEST(output_format_test)
 
   constexpr double epsilon = 1e-8;
 
-  TEST_EQUAL("metadata has same size after IO", input_md->size(), md->size());
+  EXPECT_EQ( input_md->size(), md->size() )
+    << "Metadata does not have same size after IO!";
 
   for (auto mdi : *input_md)
-  {
-    if ( !md->has( mdi.second->tag() ) )
-    {
-      TEST_ERROR("Reloaded metadata is missing tag " << mdi.second->name());
-      continue;
-    }
+  [&]{
+    ASSERT_TRUE( md->has( mdi.second->tag() ) )
```

Comment:
I realize that the anonymous lambda is used here with `ASSERT_TRUE` to accomplish what continue statement did before, but this is just not very readable.  It's not visually very easy to pick out the start and end of this closure, especially since the body is large.  Most programmers aren't going to understand what is going on here.  I like the `continue` statement better, but if there is no reasonable way to do that in GTest then at least highlight the start and end of the anonymous lambda with comments and make a comment that the `ASSERT_TRUE` exits the lambda and continues to the next step of the loop.

Reply:
Loop body has been refactored into a helper function.